runtime/include/chpl-mem.h

```diff
@@ -265,6 +265,9 @@ void chpl_mem_layerFree(void*, int32_t lineno, int32_t filename);
 #define chpl_mem_allocMany(number, size, description, lineno, filename)        \
   sys_malloc((number)*(size))
 
+#define chpl_mem_alloc(size, description, lineno, filename)        \
```

Comment:
While I sympathize with wanting this, I don't think we should have this here, and not in the non-launcher case above. Historically we just call chpl_mem_allocMany with number=1, and I'd prefer to keep doing instead of introducing the number=1 special case here.

Reply:
I don't think historical usage is a very good guide here.  As far as I've ever been able to tell the only reason the launchers were in the runtime hierarchy was to share a little bit of the runtime code.  Or really, *appear* to share, since in most cases the `.h` files define the same thing quite differently on two sides of a big `#ifndef LAUNCHER ... #else ... #endif` construct.  That's the case for `chpl_mem_allocMany()`, for example.  But then the desire to minimize the resulting special-casing in the defining .h files has led us to provide the bare minimum of support in, again for example, things like the memory interface.  And that in turn has led to stilted code: unless I missed something in my grep just now, with but one exception, every `chpl_mem_allocMany()` call in the `runtime/src/launch/` hierarchy passes `sizeof(char)` as its second argument.  That's silly.  I appreciate the desire not to do too much work in the existing launcher code because we'd like to do a wholesale replacement at some point, but until that's the most important next thing for someone to work on I don't believe it's unreasonable to make improvements, and this seems like an improvement.  (In fact, if it weren't for the fact that I would have had to test every single launcher, I would have changed all those `chpl_mem_allocMany(..., sizeof(char), ...)` calls to `chpl_mem_alloc(...)` calls also, in that last commit.)

Reply:
Man, I'm really off my reviewing game..

I had thought that you were adding a brand new`chpl_mem_alloc()` routine, but now I see that it was already defined for regular runtime (non-launcher) code, and you're just adding it to the launcher code now. Ignore me